pkg/pgmodel/end_to_end_tests/drop_test.go

```diff
@@ -452,16 +531,25 @@ func TestSQLDropAllMetricData(t *testing.T) {
 			t.Error(err)
 		}
 
-		if count != 0 {
+		if count != 1 {
 			t.Errorf("unexpected series count: %v", count)
 		}
 
-		err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.label`).Scan(&count)
+		err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.series WHERE delete_epoch IS NOT NULL`).Scan(&count)
 		if err != nil {
 			t.Error(err)
 		}
 
 		if count != 1 {
+			t.Errorf("unexpected series count: %v", count)
+		}
+
+		err = db.QueryRow(context.Background(), `SELECT count(*) FROM _prom_catalog.label`).Scan(&count)
+		if err != nil {
+			t.Error(err)
+		}
+
+		if count != 2 {
```

Comment:
we should be able to avoid restarting the ingestor at line 570 near (//Restart ingestor to avoid stale cache issues.) now that the cache issues have been solved :)

Reply:
🤔 
@cevian I might duplicate this test then; I think it's good to have a test where one ingestor deletes the Id and another one resurrects it...

Reply:
> we should be able to avoid restarting the ingestor at line 570 

It turns out this is more difficult than I thought; the Ingestor is stuck on an old epoch until it refreshes, so it cannot perform new Inserts once the epoch changes. I think our options are to add a way for the Ingestor to tell the per-metric goroutines to trash their caches (I don't love increasing the complexity of those messages, but it might be a good thing to have anyway) or just test it in the unit tests